test/integration/targets/setup_mariadb/handlers/main.yml

```diff
@@ -0,0 +1,24 @@
+- name: Remove MariaDB repo
+  yum_repository:
+    name: MariaDB
+    state: absent
+  listen: cleanup mariadb
+
+- name: Remove MariaDB related packages
+  yum:
+    name: "{{ item }}"
+    state: absent
+  loop: "{{ packages_to_cleanup }}"
```

Comment:
It's much more efficient to pass a list to the `yum` module rather than iterating over a list. This results in one task _per item in the list_ rather than one task that installs multiples packages at once.

In the past, the `yum` action plugin did some magic to avoid multiple `yum` tasks, but that has been removed.

Reply:
@samdoran , please, look https://github.com/ansible/ansible/pull/63848

Reply:
thank you for pointing this out!